src/ds18b20.toit

```diff
@@ -142,12 +149,174 @@ class Ds18b20:
+    temperature is read. Since raw temperature values have a resolution of 12
+    bits, but the alarm registers are limited to 8 bits, only bits
+    11 through 4 of the raw temperature values are used for the comparison.
+    Practically speaking, this means that the alarm temperatures can only
+    be accurate to up to 1 degree Celsius.
+
+  If the alarm functionality of the DS18B20 is not used, these registers can
+    be used to store arbitrary data.
+
+  The configuration register is a 8-bit value. Bit 7, and bits 0 through 4 are
+    reserved for internal use and can't be overwritten. Only bits 5 and 6
+    can be changed. They are used, to set the resolution of the temperature
+    readings. The resolution is set by the following table:
+
+  ```
+  | Bits 5 and 6 | Resolution | Register value |
+  | ------------ | ---------- | -------------- |
+  | 0 0          | 9 bits     | 0x1F           |
+  | 0 1          | 10 bits    | 0x3F           |
+  | 1 0          | 11 bits    | 0x5F           |
+  | 1 1          | 12 bits    | 0x7F           |
+  ```
+  */
+  write_scratchpad bytes/ByteArray --commit/bool=false:
+    if bytes.size != 3: throw "INVALID ARGUMENT"
+    if is_closed: throw "CLOSED"
+    if not bus_.reset:
+      throw "NO DEVICE FOUND"
+    select_self_
+    bus_.write_byte WRITE_SCRATCHPAD_
+    bus_.write bytes
+    if commit: copy_scratchpad_to_eeprom
+
+  /**
+  Writes the given values to the scratchpad.
+
+  The scratchpad is the DS18B20's memory. It contains the temperature, the
+    alarm temperatures, the configuration register, and the CRC. Unless
+    committed to EEPROM, the scratchpad is volatile and is reset to the
+    EEPROM values on power up. Use $commit to automatically copy the
+    scratchpad values to EEPROM. Alternatively, call
+    $copy_scratchpad_to_eeprom to copy the scratchpad values to EEPROM.
+
+  Every time the sensor measures the temperature it compares the
+    value with the $high_alarm_temperature and $low_alarm_temperature
+    values. If the temperature is outside the range, then the alarm flag
+    is set. The alarm flag is cleared automatically at the next read.
+
+  The resolution can be either 9, 10, 11, or 12 bits. The higher the
+    resolution, the longer the conversion (measurement) time:
+  - $RESOLUTION_9_BITS, 93.75 ms
+  - $RESOLUTION_10_BITS, 187.5 ms
+  - $RESOLUTION_11_BITS, 375 ms
+  - $RESOLUTION_12_BITS, 750 ms
+  */
+  write_scratchpad
+      --high_alarm_temperature/float
+      --low_alarm_temperature/float
+      --resolution/int=RESOLUTION_12_BITS
+      --commit/bool=false:
+    high_int := high_alarm_temperature.round.to_int
+    low_int := low_alarm_temperature.round.to_int
+    if not -128 <= high_int <= 127: throw "HIGH ALARM TEMPERATURE OUT OF RANGE"
+    if not -128 <= low_int <= 127: throw "LOW ALARM TEMPERATURE OUT OF RANGE"
+    if not RESOLUTION_9_BITS <= resolution <= RESOLUTION_12_BITS: throw "INVALID_ARGUMENT"
+    config_value := (12 - resolution) << 5
```

Comment:
This looks wrong.  RESOLUTION_9_BITS should give a config of 0 << 5

Reply:
oops.
done.